sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs

```diff
@@ -81,6 +81,8 @@ private static bool IsTestFailedWithRecordingMismatch(TestExecutionContext conte
 
                 return failed && context.CurrentResult.Message.Contains(typeof(TestRecordingMismatchException).FullName);
             }
+
+            private static bool IsRunningInCI() => Environment.GetEnvironmentVariable("TF_BUILD") != null;
```

Comment:
Let's throw this onto the TestEnvironment as a static property in case someone else needs it.